tests/monitoring_test.go

```diff
@@ -0,0 +1,218 @@
+package tests
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	openshiftroutev1 "github.com/openshift/api/route/v1"
+	promApi "github.com/prometheus/client_golang/api"
+	promApiv1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	promConfig "github.com/prometheus/common/config"
+	apps "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	sspv1beta1 "kubevirt.io/ssp-operator/api/v1beta1"
+	"kubevirt.io/ssp-operator/internal/common"
+	"kubevirt.io/ssp-operator/internal/operands/metrics"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/client-go/kubernetes"
+)
+
+var promClient promApiv1.API
+
+var _ = Describe("Prometheus Alerts", func() {
+	BeforeEach(func() {
+		waitUntilDeployed()
+		promClient = initializePromClient(getPrometheusUrl(*coreClient), getAuthorizationTokenForPrometheus(*coreClient))
+	})
+
+	Context("SSPCommonTemplatesModificationReverted", func() {
+		var (
+			testTemplate testResource
+		)
+		BeforeEach(func() {
+			strategy.SkipIfUpgradeLane()
+			testTemplate = createTestTemplate()
+		})
+		It("[test_id:8363] Should fire SSPCommonTemplatesModificationReverted", func() {
+			// we have to wait for prometheus to pick up the series before we increase it.
+			waitForSeriesToBeDetected(metrics.Total_restored_common_templates_increase_query)
+			expectTemplateUpdateToIncreaseTotalRestoredTemplatesCount(testTemplate)
+			waitForAlertToActivate("SSPCommonTemplatesModificationReverted")
+		})
+	})
+
+	Context("SSPFailingToReconcile Alert", func() {
+		var (
+			deploymentRes testResource
+			finalizerName = "ssp.kubernetes.io/temp-protection"
+		)
+
+		AfterEach(func() {
+			removeFinalizer(deploymentRes, finalizerName)
+			strategy.RevertToOriginalSspCr()
+			waitUntilDeployed()
```

Comment:
You are right, I did not realize that maybe the next test will assume that SSP was ready, but it is not.
This should not happen, because every test that needs SSP to be deployed, should add the call into a BeforeEach() before the test.

You can keep this line, and ignore my previous comment.